### PR TITLE
schema-keystore: Service for schema key storage for each schema path

### PR DESCRIPTION
**Pull request by zzacharo**
Signed-off-by: Zacharias Zacharodimos <zacharias.zacharodimos@cern.ch>

### CODE REVIEW

**Comment by Glignos** [comment]
LGTM

File: src/app/shared/services/schema-keys-store.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                  prop2: {
+                    type: 'string'
+                  }
+                }
+              },
+              aString: {
+                type: 'string'
+              },
+              innerArray: {
+                componentType: 'table-list',
```
Comment (jmartinm) [inline src/app/shared/services/schema-keys-store.service.spec.ts]: This is a ng2-json-editor thing

File: src/app/shared/services/schema-keys-store.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import { OrderedSet } from 'immutable';
+
+describe('SchemaKeysStoreService', () => {
+  let service: SchemaKeysStoreService;
+
+  beforeEach(() => {
+    service = new SchemaKeysStoreService();
+  });
+
+  it('should test SchemaKeysStoreService for nested complex array', () => {
```
Comment (jmartinm) [inline src/app/shared/services/schema-keys-store.service.spec.ts]: The phrase should say what the test is expecting:

`it should extract keys for each path in a nested complex array`

File: src/app/shared/services/schema-keys-store.service.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  constructor() { }
+
+  public buildSchemaKeyStore(schema: {}) {
+    this.buildSchemaKeyStoreRecursively('', schema);
+  }
+
+  private buildSchemaKeyStoreRecursively(path: string, schema: {}) {
+
+    if (schema['type'] === 'object') {
```
Comment (jmartinm) [inline src/app/shared/services/schema-keys-store.service.ts]: There seems to be quite a lot of repetition between the two branches (`object` and `array`). Would be good combine them, and for one use `schema['properties']` and for the other one `schema['items']['properties']`

File: src/app/shared/services/schema-keys-store.service.spec.ts (hunk 1 above)
Comment (zzacharo) [reply src/app/shared/services/schema-keys-store.service.spec.ts]: leftover.

**Comment by zzacharo** [comment]
@jmartinm check again!

**Comment by zzacharo** [comment]
@jmartinm ping!

**Comment by zzacharo** [comment]
@jmartinm this is open yet! Can you have a final look and merge it? Or I can merge it!

**Comment by jmartinm** [comment]
Forgot about this, merged now ;)

**Pull request merged**